doc/project/checklist.md

```diff
@@ -395,3 +395,91 @@ Clean up all compile-time and run-time warnings thrown by the simulator.
 ### PRE_VERIFIED_SUB_MODULES_V3
 
 Sub-modules that are pre-verified with their own testbenches have already reached V3 stage.
+
+## S1
+
+For a transition from S0 to S1, the following items are expected be completed.
+
+### DIF_EXISTS
+
+`dif_<ip>.h` and, optionally, `dif_<ip>.c` exist in `sw/device/lib/dif`.
+
+### DIF_USED_IN_TREE
+
+All existing in-tree code which uses the device, uses the device via the DIF.
+There is no remaining driver code that directly uses the device outside of DIF code.
+
+### DIF_TEST_UNIT
+
+Unit tests exist for the DIF in `sw/device/tests/dif` named `dif_<ip>_unittest.cc`.
```

Comment:
You are right that the current testing implementation is mock testing. However, @gkelly and I do not want to commit ourselves to mock testing forever, so we've phrased it as "unit tests", which is a software testing approach, rather than "mocked tests" which is an implementation technique.

Reply:
We don't have that documentation yet, but we can look to adding it in the future.